mailet/icalendar/src/main/java/org/apache/james/transport/mailets/ICALToHeader.java

```diff
@@ -82,37 +88,41 @@ public String getMailetInfo() {
 
     @Override
     public void init() throws MessagingException {
-        attribute = getInitParameter(ATTRIBUTE_PROPERTY, ATTRIBUTE_DEFAULT_NAME);
+        attribute = getInitParameter(CONFIG_ATTRIBUTE_PROPERTY, CONFIG_ATTRIBUTE_DEFAULT_NAME);
         if (Strings.isNullOrEmpty(attribute)) {
             throw new MessagingException("Attribute " + attribute + " can not be empty or null");
         }
     }
 
     @Override
     public void service(Mail mail) throws MessagingException {
-        if (mail.getAttribute(attribute) == null) {
+        if (!mail.getAttribute(AttributeName.of(attribute)).isPresent()) {
             return;
         }
         try {
             getCalendarMap(mail)
-                .values()
-                .stream()
-                .findAny()
-                .ifPresent(Throwing.<Calendar>consumer(calendar -> writeToHeaders(calendar, mail))
-                    .sneakyThrow());
+                .ifPresent(calendarMap -> writeFirstValueFromCalendarMap(mail, calendarMap));
         } catch (ClassCastException e) {
             LOGGER.error("Received a mail with {} not being an ICAL object for mail {}", attribute, mail.getName(), e);
         }
     }
 
+    public void writeFirstValueFromCalendarMap(Mail mail, Map<String, Calendar> calendarMap) {
+        calendarMap
+            .values()
+            .stream()
+            .findAny()
+            .ifPresent(Throwing.<Calendar>consumer(calendar -> writeToHeaders(calendar, mail)).sneakyThrow());
+    }
+
     @VisibleForTesting
     String getAttribute() {
         return attribute;
     }
 
     @SuppressWarnings("unchecked")
-    private Map<String, Calendar> getCalendarMap(Mail mail) {
-        return (Map<String, Calendar>) mail.getAttribute(attribute);
+    private Optional<Map<String, Calendar>> getCalendarMap(Mail mail) {
+        return AttributeUtils.getValueAndCastFromMail(mail, AttributeName.of(attribute), (Class<Map<String, Calendar>>)(Object) Map.class);
```

Comment:
Can we extract `(Class<Map<String, Calendar>>)(Object) Map.class` to a `MAP_OF_STRING_TO_CALENDAR_CLASS` constant?

It would make reading so much easier...